packages/synapse-react-client/src/components/Forum/DiscussionReply.tsx

```diff
@@ -15,27 +15,29 @@ import { displayToast } from '../ToastMessage/ToastMessage'
 import { UserBadge } from '../UserCard/UserBadge'
 import { ForumThreadEditor } from './ForumThreadEditor'
 import { Box } from '@mui/material'
+import { copyStringToClipboard } from '@/utils/functions/StringUtils'
 
 export type DiscussionReplyProps = {
   reply: DiscussionReplyBundle
   isReplyAuthorModerator?: boolean
   onClickLink?: () => void
 }
 
-const DEFAULT_ON_CLICK_LINK = () =>
-  alert('This functionality has not been implemented yet')
+const handleCopyLink = (id: string) => {
+  const baseUrl = `${window.location.origin}${window.location.pathname}`
+  const url = `${baseUrl}?replyid=${id}`
+  copyStringToClipboard(url)
+  displayToast('Reply link copied to clipboard', 'info')
```

Comment:
`copyStringToClipboard` is async
```suggestion
  copyStringToClipboard(url).then(() => displayToast('Reply link copied to clipboard', 'info'))
```